src/app/core/services/opa/service.ts

```diff
@@ -61,4 +63,36 @@ export class OPAService {
     const url = `${this._newRestRoot}/constrainttemplates/${ctName}`;
     return this._http.delete(url);
   }
+
+  constraints(projectId: string, clusterId: string): Observable<Constraint[]> {
+    this._constraints$ = merge(this._constraintsRefresh$, this._refreshTimer$)
```

Comment:
`this._constraints$` is assigned each time the function is invoked, why?

Reply:
If this should have the role of a cache then it should look like admin settings getter for example. If not, then `this._constraints$` is not required and you can return directly.

Reply:
I'd add cache here similar to what I've done in https://github.com/kubermatic/dashboard/pull/3033/files. This will make sure that we do not create multiple observables for the same projectID/clusterID pair.

Reply:
Good catch. Wanted to cache the data, but forgot it half of the way. 🙈